pkgs/development/libraries/libvirt/default.nix

```diff
@@ -66,45 +66,46 @@ in stdenv.mkDerivation rec {
     substituteInPlace src/lxc/lxc_conf.c \
       --replace 'lxc_path,' '"/run/libvirt/nix-emulators/libvirt_lxc",'
     patchShebangs . # fixes /usr/bin/python references
-    mkdir -p build && cd build
   '';
 
-  configureScript = "../configure";
-
   dontAddDisableDepTrack = true;
 
-  configureFlags = [
-    "--with-runstatedir=/run" # TODO: remove when autoconf 2.70 is released
-    "--localstatedir=/var"
-    "--sysconfdir=/var/lib"
-    "--with-libpcap"
-    "--with-qemu"
-    "--with-vmware"
-    "--with-vbox"
-    "--with-test"
-    "--with-esx"
-    "--with-remote"
-    "--with-polkit"
+  mesonFlags = let
+    opt = option: enable: "-D${option}=${if enable then "enabled" else "disabled"}";
+  in [
+    "-Dauto_features=disabled"
+    "-Drunstatedir=/run"
+    "-Dlocalstatedir=/var"
+    "-Dsysconfdir=/var/lib"
+    "-Dlibpcap=enabled"
+    "-Ddriver_qemu=enabled"
+    "-Ddriver_vmware=enabled"
+    "-Ddriver_vbox=enabled"
+    "-Ddriver_test=enabled"
+    "-Ddriver_esx=enabled"
+    "-Ddriver_remote=enabled"
+    "-Dpolkit=enabled"
+    "-Dbus=enabled"
+    "-Dfuse=disabled" # todo check against previous config
+    "-Dhal=disabled" # todo check against previous config
+    "-Dglusterfs=disabled" # todo check against previous config
+    (opt "storage_iscsi" enableIscsi)
+    (opt "storage_zfs" (stdenv.isLinux && zfs != null))
   ] ++ optionals stdenv.isLinux [
-    "QEMU_BRIDGE_HELPER=/run/wrappers/bin/qemu-bridge-helper"
-    "QEMU_PR_HELPER=/run/libvirt/nix-helpers/qemu-pr-helper"
```

Comment:
@jtojnar any idea how to set those variables now ?  I've seen they still exist in the meson build but doesn't seem like we can set them, more like we'll have to susbstitute them afterwards

Reply:
Yeah, this is not possible with Meson/`ninja`. You will need to patch it here https://gitlab.com/libvirt/libvirt/-/blob/v6.7.0/meson.build#L1818-1827.

Reply:
I used sed, it looks ok